draft-ietf-lake-edhoc.md

```diff
@@ -650,59 +654,52 @@ where salt = 0x (zero-length byte string).
-
-### PRK_3m
+If the Responder authenticates with a static Diffie-Hellman key, then PRK_3e2m = Extract( SALT_3e2m, G_RX ), where
 
-The pseudo-random key PRK_3m is used to produce a MAC in message_3. PRK_3m is derived as follows:
+* SALT_3e2m is derived from PRK_2e, see {{expand}}, and
+* G_RX is the ECDH shared secret calculated from G_R and X, or G_X and R (the Responder's private authentication key, see {{auth-keys}}),
 
-If the Initiator authenticates with a static Diffie-Hellman key, then PRK_3m = Extract( PRK_3e2m, G_IY ), where G_IY is the ECDH shared secret calculated from G_I and Y, or G_Y and I (the Initiator's private authentication key, see {{auth-keys}}), else PRK_3m = PRK_3e2m.
+else PRK_3e2m = PRK_2e.
 
-### PRK_out
+### PRK_4e3m
 
-The pseudo-random key PRK_out is used to derive application specific data and keying material to encrypt message_4. PRK_out is derived as follows:
+The pseudo-random key PRK_4e3m is used to produce a MAC in message_3 and to encrypt message_4. PRK_4e3m is derived as follows:
 
-PRK_out = Extract(PRK_3m, TH_4)
+If the Initiator authenticates with a static Diffie-Hellman key, then PRK_4e3m = Extract( SALT_4e3m, G_IY ), where
 
-The transcript hash TH_4 is a CBOR encoded bstr and the input to the hash function is a CBOR Sequence.
+* SALT_4e3m is derived from PRK_3e2m, see {{expand}}, and
+* G_IY is the ECDH shared secret calculated from G_I and Y, or G_Y and I (the Initiator's private authentication key, see {{auth-keys}}),
 
-~~~~~~~~~~~
-   TH_4 = H( TH_3, PLAINTEXT_3 )
-~~~~~~~~~~~
+else PRK_4e3m = PRK_3e2m.
 
-where H() is the EDHOC hash algorithm in the selected cipher suite.
 
 ## Expand {#expand}
 
-The keys, IVs and MACs used in EDHOC are derived from the PRKs using Expand, and instantiated with the EDHOC AEAD algorithm in the selected cipher suite.
+The keys, IVs, salts, and MACs used in EDHOC are derived from the PRKs using Expand.
 
 ~~~~~~~~~~~~~~~~~~~~~~~
-   OKM = EDHOC-KDF( PRK, transcript_hash, label, context, length )
+   OKM = EDHOC-KDF( PRK, label, context, length )
        = Expand( PRK, info, length )
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 where info is encoded as the CBOR sequence
 
 ~~~~~~~~~~~ CDDL
 info = (
-  transcript_hash : bstr,
-  label : tstr,
+  label : uint,
   context : bstr,
   length : uint,
 )
 ~~~~~~~~~~~
 
 where
 
-  + transcript_hash is a bstr set to one of the transcript hashes TH_2, TH_3, or TH_4 as defined in Sections {{asym-msg2-form}}{: format="counter"}, {{asym-msg3-form}}{: format="counter"}, and {{exporter}}{: format="counter"}.
-
-  + label is a tstr set to the name of the derived key, IV or MAC; i.e., "KEYSTREAM_2", "MAC_2", "K_3", "IV_3", or "MAC_3".
+  + label is a uint
 
-  + context is a bstr
+  + context is a bstr typically including one of the transcript hashes TH_2, TH_3, or TH_4 as defined in Sections {{asym-msg2-proc}}{: format="counter"} and {{asym-msg3-proc}}{: format="counter"}.
 
```

Comment:
Remove "ypically including one of the transcript hashes TH_2, TH_3, or TH_4"
This is not true for the exporter

Reply:
Moved text about TH to paragraph below